content/gotraining.slide

```diff
@@ -740,11 +748,108 @@ Use `gofmt` or `goimport` as part of your editor/IDE setup
 [[https://godoc.org/gopkg.in/square/go-jose.v2][go-jose]] - JWT utils
  ...
 
-* Project work
+
+* A RESTful API starter (1/3)
+
+[[https://goplay.space/#oPHSAHa_Zo1][Full basic sample]]
+[[https://goplay.space/#ddCcD-1JdDJ][Full sample]] with error handling
+
+	type TeapotHandler interface {
+		handlePost(w http.ResponseWriter, r *http.Request)
+		handleGet(w http.ResponseWriter, r *http.Request)
+	}
+
+	func SetupRoutes(h TeapotHandler, r chi.Router) {
```

Comment:
Maybe swap the function args so the router is first, given this is `SetupRoutes` ? I don't know if there is an idiomatic order here though.

Reply:
Done.

Reply:
This was done in the sample, but not the slides.

Reply:
Donedone.